### ISSUE TITLE
Warnings due to `@frozen` attribute on non-public enums with Xcode 16 Beta 3

### ISSUE THREAD

**Issue by ojun9** [apple/swift-openapi-generator#586]
### Description

Using Xcode 16 Beta 3, the generated `Type.swift` file now shows multiple warnings due to the `@frozen` attribute being applied to non-public enums. The `Type.swift` file is generated with non-public enums that include the `@frozen` attribute. For example:

```swift
@frozen internal enum Output: Sendable, Hashable { /* ... */ }
```

This results in the following warning:

```shell
@frozen has no effect on non-public enums
```

### Reproduction

To reproduce the issue, it is necessary to generate the `Type.swift` file.

**openapi-generator-config.yaml**
```yaml
generate:
  - types
  - client
```

### Package version(s)

`1.2.1`

### Expected behavior

The `@frozen` attribute should not be applied to non-public enums to avoid unnecessary warnings.

### Environment

- Swift version Info: 
  - Apple Swift version 6.0 (swiftlang-6.0.0.5.15 clang-1600.0.22.6)
- Xcode version Info: 
  - Xcode 16 Beta 3

### Additional information

This issue arises from a new warning introduced in Xcode 16 Beta 3.

**Comment by czechboy0** [comment]
Thanks for reporting this, @ojun9! I think the fix here is what you suggested - only generate the frozen attribute when the access modifier is public or package. Would you consider opening a PR for that?

**Comment by ojun9** [comment]
Thank you for the feedback! 
I’ll work on the fix over the weekend and open a PR to ensure the frozen attribute is only generated when the access modifier is public or package.

**Comment by czechboy0** [comment]
Looks like #587 is already fixing this issue

**Comment by ojun9** [comment]
@czechboy0

Since there were no changes in the PR https://github.com/apple/swift-openapi-generator/pull/587, I created a new PR https://github.com/apple/swift-openapi-generator/pull/595. Please let me know if there are any issues!

**Comment by czechboy0** [comment]
Fixed and released in 1.3.0: https://github.com/apple/swift-openapi-generator/releases/tag/1.3.0
